client/my-sites/plans-grid/components/header-price.tsx

```diff
@@ -147,7 +154,7 @@ const PlanFeatures2023GridHeaderPrice = ( {
 		<>
 			{ isPricedPlan ? (
 				<HeaderPriceContainer>
-					{ shouldShowIntroPricing && (
+					{ shouldShowIntroPricing ? (
```

Comment:
`Non-blocking / Nitpick`:

The approach ( toggling badge CSS visibility ) looks good to me. One thing that threw me off a little was the level of nesting ( at it's deepest, we check `isPricedPlan` -> `! shouldShowIntroPricing` -> `! shouldShowDiscountedPrice` -> `isAnyVisibleGridPlanDiscounted` ) It was a bit confusing to wrap my head around at first glance.

Not the end of the world, and I'd much prefer we get the fix out for this not-so-appealing design issue first, but wondering if we feel like this is worth taking a closer look at.

Reply:
It's a legitimate concern. Ultimately, these are nested ternaries that we would have blocked otherwise.
FWIW the original wasn't any better, just flatter by checking on `! shouldShowIntroPricing` in every condition. 

I think there are bigger issues with the implementation, keeping prices and badges together like this isn't ideal (hence this hack that wouldn't be necessary otherwise). They could have been separated, but that won't be easy to do. 

I can revert this back to the flatter conditioning though

Reply:
> It's a legitimate concern. Ultimately, these are nested ternaries that we would have blocked otherwise.
FWIW the original wasn't any better, just flatter by checking on ! shouldShowIntroPricing in every condition.

Ah I see the tradeoff -- thanks for explaining

> I think there are bigger issues with the implementation, keeping prices and badges together like this isn't ideal (hence this hack that wouldn't be necessary otherwise). They could have been separated, but that won't be easy to do.

I'm not super familiar with the header-price component or how it got to where it is now, but I do agree that it feels like some sort of code smell. I'm also not the biggest fan of encouraging the early return to grow 🤔

> cleaned this up a bit. There's only a one-level deep ternary now in the last block.

The code now, as a whole, does feel easier to read for me. This component could probably use some love, but a refactor sounds like a bigger lift, and unfortunately I don't have any other ideas for besides that for now. Thanks for providing the context though